projects/core/src/auth/user-auth/facade/auth.service.ts

```diff
@@ -49,6 +47,10 @@ export class AuthService {
    */
   logoutInProgress$: Observable<boolean> = new BehaviorSubject<boolean>(false);
 
+  protected csrfToken$ = this.crossSiteRequestForgeryService
```

Comment:
why do we break it down into method and property?

Reply:
This is so that the shareReplay takes effect for multiple subscribers.  If we create a new observable each time with shareReplay, each instance will make its own request and shareReplay instance.